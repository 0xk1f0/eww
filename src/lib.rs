pub mod ast;
use lalrpop_util::lalrpop_mod;

lalrpop_mod!(pub parser);

macro_rules! test_parser {
    ($($text:literal),*) => {{
        let p = crate::parser::ExprParser::new();
        //use crate::lexer::Lexer;

        ::insta::with_settings!({sort_maps => true}, {
            $(
                ::insta::assert_debug_snapshot!(p.parse($text));
            )*
        });
    }}
}

#[cfg(test)]
mod tests {
    #[test]
    fn test() {
        test_parser!(
            "1",
            "2 + 5",
            "2 * 5 + 1 * 1 + 3",
            "(1 + 2) * 2",
            "1 + true ? 2 : 5",
            "1 + true ? 2 : 5 + 2",
            "1 + (true ? 2 : 5) + 2",
            "foo(1, 2)",
            "! false || ! true"
        );
    }
}
