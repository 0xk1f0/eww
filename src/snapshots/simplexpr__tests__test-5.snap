---
source: src/lib.rs
expression: "p.parse(\"1 + true ? 2 : 5\")"

---
Ok(
    IfElse(
        BinOp(
            Literal(
                "1",
            ),
            Plus,
            Literal(
                "true",
            ),
        ),
        Literal(
            "2",
        ),
        Literal(
            "5",
        ),
    ),
)
